### ISSUE TITLE
Feature request: Useless cat when piping

### ISSUE THREAD

**Issue by ember91** [koalaman/shellcheck#2793]
#### For new checks and feature suggestions
- [ X ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ X ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
wc -l file | cat
```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

Warning: Useless cat. Do like this:
```sh
wc -l file
```

**Comment by aeiplatform** [comment]
ShellCheck isn't static analysis tool for detecting errors between chair and keyboard.

**Comment by ember91** [comment]
That's not true. Shellcheck already detects a useless cat when doing e.g. cat file | grep ... instead of grep ... file. Although this is only a suggestion.

**Comment by wileyhy** [comment]
"...between chair and keyboard" could easily be read as, "all errors," you
know. It is somewhat vague.

**Comment by koalaman** [comment]
"Useless Use Of Cat" is/was a meme in the shell scripting community, predating ShellCheck by decades. The warning is about 80% nod to tradition and 20% helpful advice.

While classic UUOC happens all the time organically, it's much more rare for anyone to unintentionally append `| cat` to anything. If they do, it's much more likely to be for semi-legitimate reasons like making stdout not a tty for formatting and buffering reasons.

Looking across a corpus, the most common case (which is still quite rare compared to `cat file | ..`) is to have a `cat` seemingly added to make a tool stop outputting in color, but which was subsequently made redundant by redirecting or capturing the output. This would be a nifty thing to point out, but it's ultimately very low impact, so it's likely not worth the complexity.

Thanks for the suggestion!
